e2e/stack.go

```diff
@@ -0,0 +1,213 @@
+	return &Stack{
+		l1Address:        l1Address,
+		monomerAddress:   monomerAddress,
+		opNodeAddress:    opNodeAddress,
+		contractsRootDir: contractsRootDir,
+		eventListener:    eventListener,
+		l1BlockTime:      l1BlockTime,
+	}
+}
+
+func (s *Stack) Run(parentCtx context.Context) (err error) {
+	var wg conc.WaitGroup
+	defer wg.Wait()
+	ctx, cancel := context.WithCancelCause(parentCtx)
+	defer func() {
+		cancel(err)
+		err = utils.Cause(ctx)
+	}()
+
+	const numConnectAttempts = 20
+
+	// Run anvil.
+	_, l1Port, err := net.SplitHostPort(s.l1Address)
+	if err != nil {
+		return fmt.Errorf("split l1 address into host and port: %v", s.l1Address)
+	}
+	wg.Go(func() {
+		cancel(fmt.Errorf("run cmd: %v", s.runCmd(
+			ctx,
+			"anvil",
+			"--port", l1Port,
+			"--order", "fifo",
+			"--disable-block-gas-limit",
+			"--gas-price", "0",
+			"--block-time", fmt.Sprint(s.l1BlockTime.Seconds()),
+		)))
+	})
+	if !utils.AddressIsReady(s.l1Address, numConnectAttempts) {
+		return fmt.Errorf("could not connect to anvil at address %s", s.l1Address)
+	}
+
+	// Fund an account.
+	anvilHTTPAddress := "http://" + s.l1Address
+	anvilRPCClient, err := rpc.DialContext(ctx, anvilHTTPAddress)
+	if err != nil {
+		return fmt.Errorf("dial anvil: %v", err)
+	}
+	anvil := NewAnvil(anvilRPCClient)
+	privKey, err := crypto.GenerateKey()
+	if err != nil {
+		return fmt.Errorf("generate key: %v", err)
+	}
+	address := crypto.PubkeyToAddress(privKey.PublicKey)
+	if err := anvil.SetBalance(ctx, address, 1000000000000000000); err != nil { //nolint:gomnd
+		return fmt.Errorf("set balance: %v", err)
+	}
+
+	// Deploy L1 contracts.
+	if err := s.runCmd(
+		ctx,
+		"forge",
+		"script",
+		"--root", s.contractsRootDir,
+		"-vvv",
+		fmt.Sprintf("%s:Deploy", filepath.Join(s.contractsRootDir, "scripts", "Deploy.s.sol")),
+		"--rpc-url", anvilHTTPAddress,
+		"--broadcast",
+		"--private-key", common.Bytes2Hex(crypto.FromECDSA(privKey)),
+	); err != nil {
+		return fmt.Errorf("deploy op L1 contracts: %v", err)
+	}
+	latestL1Block, err := anvil.BlockByNumber(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("get the latest l1 block: %v", err)
+	}
+
+	// Run Monomer.
+	const l2ChainID = 901
+	wg.Go(func() {
+		cancel(fmt.Errorf("run monomer: %v", runMonomer(ctx, latestL1Block.Time(), l2ChainID, s.monomerAddress)))
+	})
+	if !utils.AddressIsReady(s.monomerAddress, numConnectAttempts) {
+		return fmt.Errorf("waiting for monomer websocket server at address %s", s.monomerAddress)
+	}
+	monomerRPCClient, err := rpc.DialContext(ctx, "ws://"+s.monomerAddress)
+	if err != nil {
+		return fmt.Errorf("dial monomer: %v", err)
+	}
+	monomerClient := NewMonomer(monomerRPCClient)
+	l2GenesisBlockHash, err := monomerClient.GenesisHash(ctx)
+	if err != nil {
+		return fmt.Errorf("get Monomer genesis block hash: %v", err)
+	}
+
+	// Get deploy config and rollup config.
+	const networkName = "hardhat"
```

Comment:
?

```suggestion
	const networkName = "anvil"
```

Reply:
I'll have to add a comment for this; it's deliberate. I forget the details at this moment, but anvil is meant to be a drop-in replacement for hardhat in the op scripts IIRC.

Reply:
:+1: 

Thought there was a chance this was true, but I recalled you had swapped backends a couple of times during development and figured it might have been a leftover.